src/core/grid_based_algorithms/lb.cpp

```diff
@@ -2643,74 +2600,14 @@ inline void lb_viscous_coupling(Particle *p, double force[3]) {
   v_drift += p->p.mu_E;
 #endif
 
-  force[0] = -lbpar.friction * (velocity[0] - v_drift[0]);
-  force[1] = -lbpar.friction * (velocity[1] - v_drift[1]);
-  force[2] = -lbpar.friction * (velocity[2] - v_drift[2]);
-
-  force[0] = force[0] + p->lc.f_random[0];
-  force[1] = force[1] + p->lc.f_random[1];
-  force[2] = force[2] + p->lc.f_random[2];
-
-  /* transform momentum transfer to lattice units
-     (Eq. (12) Ahlrichs and Duenweg, JCP 111(17):8225 (1999)) */
-  delta_j[0] = -force[0] * time_step * lbpar.tau / lbpar.agrid;
-  delta_j[1] = -force[1] * time_step * lbpar.tau / lbpar.agrid;
-  delta_j[2] = -force[2] * time_step * lbpar.tau / lbpar.agrid;
-
-  lattice_interpolation(lblattice, p->r.p,
-                        [&delta_j](Lattice::index_t index, double w) {
-                          auto &node = lbfields[index];
-
-                          node.force_density[0] += w * delta_j[0];
-                          node.force_density[1] += w * delta_j[1];
-                          node.force_density[2] += w * delta_j[2];
-                        });
-
-  // map_position_to_lattice: position ... not inside a local plaquette in ...
+  /* calculate viscous force
+   * (Eq. (9) Ahlrichs and Duenweg, JCP 111(17):8225 (1999))
+   * */
+  auto const force = -lbpar.friction * (p->m.v - v_drift) + f_random;
 
-#ifdef ENGINE
-  if (p->swim.swimming) {
-    // TODO: Fix LB mapping
-    if (n_nodes > 1) {
-      if (this_node == 0) {
-        fprintf(stderr, "ERROR: Swimming is not compatible with Open MPI and "
-                        "CPU LB on more than 1 node.\n");
-        fprintf(stderr, "       Please use LB_GPU instead.\n");
```

Comment:
Is this limitation removed now?

Reply:
Yes, as far as I can tell. The existing test passes now with multiple nodes, and conceptually it should work.